### DISCUSSION TITLE
Does RTK supports Flow (typed) ?

### DISCUSSION THREAD

**Discussion by Yohandah** [reduxjs/redux-toolkit#3574] [Q&A] [answered]
My question is does RTK officially supports Flow typing ? I'm trying to use RTK with flow and it seems it doesn't have type check

<img width="1269" alt="image" src="https://github.com/reduxjs/redux-toolkit/assets/15921402/32f08c51-886c-40f1-adbf-073db71b40ea">

<img width="921" alt="image" src="https://github.com/reduxjs/redux-toolkit/assets/15921402/5e4f50ea-a680-45ef-8a69-4cefc0c0433a">

**Comment by EskiMojo14** [comment]
RTK is written with Typescript, I'm not sure how well Flow does with Typescript declaration files.

Though for what it's worth, you shouldn't ever be using the AsyncThunk type directly like that - the return type from createAsyncThunk should already be accurate. See [Usage with Typescript](https://redux-toolkit.js.org/usage/usage-with-typescript#createasyncthunk)

**Reply by Yohandah** [reply]
Yes I know I shouldn't but Flow tells me to ... 😢 So I guess RTK is not compatible with Flow ?

<img width="1236" alt="image" src="https://github.com/reduxjs/redux-toolkit/assets/15921402/48452afd-b0b4-4fc0-868f-f4ab3fb61021">

**Reply by EskiMojo14** [reply]
is there any reason you're using Flow over Typescript, out of curiosity?

**Reply by Yohandah** [reply]
~8 years old project at work with a huge code base that is really complicated to migrate to TS :/

**Reply by Yohandah** [reply]
@EskiMojo14  So you confirm there's no support for Flow then ?

**Reply by EskiMojo14** [reply, marked as answer]
To my knowledge we have nothing specifically to support Flow, as the library is written in TS and compiled down to JS with TS declaration files.

I have no experience with Flow, so I don't know if the Typescript support at all helps with it.

**Reply by Yohandah** [reply]
Thanks for your quick response. It [can help](https://github.com/joarwilk/flowgen) but yes that means no support.

**Reply by Yohandah** [reply]
There's a [definition file in flow-typed](https://github.com/flow-typed/flow-typed/blob/master/definitions/npm/%40reduxjs/toolkit_v1.x.x/flow_v0.104.x-/toolkit_v1.x.x.js) (community driven types) but it's not complete, for instance `createAsyncThunk` is missing. I'll just keep redux vanilla I guess
